lib/chewy/journal.rb

```diff
@@ -123,21 +127,60 @@ def type_name
         JOURNAL_MAPPING.keys.first
       end
 
-      def query(time, comparator, query_type = :filter)
-        {
-          query: {
-            filtered: {
-              query_type => {
-                range: {
-                  created_at: {
-                    comparator => time.to_i
+      def query(time, comparator, index = nil, use_filter = true)
+        filter_query =
+          if use_filter
```

Comment:
maybe it makes sense to move some logic to separate methods? as for me, there are too many `if-else` statements in this method. Also, I would move this part to a separate method instead of just assigning it to `filter_query`